src/partition/ReceivedBoundingBox.hpp

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include "Partition.hpp"
+#include "logging/Logger.hpp"
+#include <vector>
+#include "mesh/Vertex.hpp"
+#include "mesh/Mesh.hpp"
+
+
+// Forward delcration to friend the boost test struct
+
+namespace PartitionTests {
+namespace ReceivedBoundingBoxTests {
+struct TestReceivedBoundingBox2D;
+struct TestReceivedBoundingBox3D;
+struct TestM2NMeshExchange;
+}
+}
+
+namespace PartitionTests {
+namespace ProvidedBoundingBoxTests {
+struct TestProvidedBoundingBox2D;
+struct TestProvidedBoundingBox3D;
+struct TestM2NMeshExchange;
+}
+}
+
+namespace precice {
+namespace partition {
+/**
+ * @brief A partition that is computed from a mesh received from another participant.
+ *
+ * A mesh is received by the master rank and re-partitioned among all slave ranks.
+ * Afterwards necessary distribution data structures are set up.
+ */
+class ReceivedBoundingBox : public Partition
+{
+public:
+
+  /// Defines the typ of geometric filter used
+  enum GeometricFilter {
```

Comment:
Why is this needed here?

Reply:
We can use Broadcast_Filter in the bb method, filter_first does not work, I kept it so we can choose to filter after broadcast or not. filter_first can be deleted!

Reply:
But we don't filter any vertices here, right? Is this already something we need in a later WP?

Reply:
yes, but we need that for next WP. As its an argument of the constructor, I keep it!

Reply:
I know this is pedantic, but the PR will be much easier to read if it is really restricted to the first WP only. Also the constructor can be modified ;)